webapp/channels/src/components/edit_post/edit_post.tsx

```diff
@@ -1,671 +1,42 @@
-
-    let emojiPicker = null;
-
-    if (config.EnableEmojiPicker === 'true') {
-        emojiPicker = (
-            <>
-                <EmojiPickerOverlay
-                    show={showEmojiPicker}
-                    target={getEmojiTargetRef}
-                    onHide={hideEmojiPicker}
-                    onEmojiClick={handleEmojiClick}
-                    onGifClick={handleGifClick}
-                    enableGifPicker={config.EnableGifPicker === 'true'}
-                    topOffset={TOP_OFFSET}
-                    rightOffset={RIGHT_OFFSET}
-                />
-                <button
-                    aria-label={formatMessage({id: 'emoji_picker.emojiPicker.button.ariaLabel', defaultMessage: 'select an emoji'})}
-                    id='editPostEmoji'
-                    ref={emojiButtonRef}
-                    className='style--none post-action'
-                    onClick={toggleEmojiPicker}
-                >
-                    <EmoticonPlusOutlineIcon
-                        size={18}
-                        color='currentColor'
-                    />
-                </button>
-            </>
-        );
+    if (!editingPostDetailsAndPost.show) {
+        return null;
     }
 
-    let rootId = '';
-    if (editingPost.post) {
-        rootId = editingPost.post.root_id || editingPost.post.id;
-    }
+    const channelId = editingPostDetailsAndPost.post.channel_id;
+    const location = editingPostDetailsAndPost.isRHS ? Locations.RHS_COMMENT : Locations.CENTER;
+    const rootId = editingPostDetailsAndPost.post.root_id || '';
+    const storageKey = `${StoragePrefixes.EDIT_DRAFT}${editingPostDetailsAndPost.post.id}`;
 
     return (
-        <div
-            className={classNames('post--editing__wrapper', {
-                scroll: renderScrollbar,
-            })}
-            ref={wrapperRef}
-        >
-            <Textbox
-                tabIndex={0}
-                rootId={rootId}
-                onChange={handleChange}
-                onKeyPress={handleEditKeyPress}
-                onKeyDown={handleKeyDown}
-                onBlur={handleBlur}
-                onHeightChange={handleHeightChange}
-                handlePostError={handlePostError}
-                onPaste={handlePaste}
-                value={editText}
+        <div className='post-edit__container'>
+            <AdvancedTextEditor
+                location={location}
                 channelId={channelId}
-                emojiEnabled={config.EnableEmojiPicker === 'true'}
-                createMessage={formatMessage({id: 'edit_post.editPost', defaultMessage: 'Edit the post...'})}
-                supportsCommands={false}
-                suggestionListPosition='bottom'
-                id='edit_textbox'
-                ref={textboxRef}
-                characterLimit={rest.maxPostSize}
-                useChannelMentions={rest.useChannelMentions}
+                postId={rootId}
```

Comment:
@hmhealey do you know why we are passing root ID here instead of post.id?

Reply:
Passing the root ID is actually correct. The name of this field is misleading. I'm going to make a followup ticket to rename that since I know this has confused me and Zubair at different points as well

You mentioned in a DM that it's causing issues when editing the first post of a thread. Looking at this code, I'd guess that that might be because `rootId` is being set to `post.root_id || post.id` above when it should just be set to `post.root_id ?? ''

Reply:
Actually setting it to `post.root_id || post.id` fixed the issue. Do you think this can cause some problem? So far I tested and everything works well.